tv_apps/browser/js/bookmarkStore.js

```diff
@@ -1,34 +1,109 @@
-/* globals BrowserDB */
+/* globals BrowserDB, SyncBrowserDB, SyncManagerBridge */
 'use strict';
 
 (function(exports){
+
   var BookmarkStore = {
+    isSynced: false,
+
     cache: [],
 
+    currentFolder : null,
+
+    reset: function(cb) {
+      this.cache = [];
+      this.currentFolder = null;
+      SyncManagerBridge.getInfo().then(message => {
+        this.isSynced = (message.state === 'enabled') ? true : false;
+        cb();
+      });
+    },
+
     getByRange: function(start, num, folderId, cb) {
-      var i = start,
+      var fn = function(){
+        var i = start,
           length = (start + num) > this.cache.length ?
```

Comment:
Sorry that I missed this part last time. Please separate ```length``` and ```result``` with another 2 ```var``` definitions. For better readability.